src/par_iter/map.rs

```diff
@@ -21,17 +21,15 @@ impl<M, MAP_OP, R> ParallelIterator for Map<M, MAP_OP>
 {
     type Item = R;
 
-    fn drive_unindexed<'c, C: UnindexedConsumer<'c, Item=Self::Item>>(self,
-                                                                      consumer: C,
-                                                                      shared: &'c C::Shared)
-                                                                      -> C::Result {
-        let consumer1: MapConsumer<M::Item, C, MAP_OP> = MapConsumer::new(consumer);
-        let shared1 = (shared, &self.map_op);
-        self.base.drive_unindexed(consumer1, &shared1)
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        let consumer1 = MapConsumer::new(consumer, &self.map_op);
+        self.base.drive_unindexed(consumer1)
     }
 }
 
-unsafe impl<M, MAP_OP, R> BoundedParallelIterator for Map<M, MAP_OP>
+impl<M, MAP_OP, R> BoundedParallelIterator for Map<M, MAP_OP>
     where M: BoundedParallelIterator,
           MAP_OP: Fn(M::Item) -> R + Sync,
```

Comment:
Oh wait, dw - it must be the return value, because `R` is a type parameter, not a trait. :P